scvi/module/_mixupvae.py

```diff
@@ -435,18 +469,21 @@ def generative(
         z_pseudobulk,
         library,
         library_pseudobulk,
-        batch_index,
+        one_hot_batch_index,
+        one_hot_batch_index_pseudobulk,
         cont_covs=None,
+        cont_covs_pseudobulk=None,
         categorical_input=(),
         categorical_pseudobulk_input=(),
+        pseudobulk_indices=(),
         size_factor=None,
         y=None,
         transform_batch=None,
     ):
         """Runs the generative model."""
         if self.pseudo_bulk == "post_inference":
             # create it from z by overwritting the one coming from inference
-            z_pseudobulk = z.mean(axis=0).unsqueeze(0)
+            z_pseudobulk = z[pseudobulk_indices, :].mean(axis=1)
```

Comment:
I don"t remember the reason why we are overwriting `z_pseudobulk` here ..

Reply:
It's because if we want to create the pseudobulk post-inference (and not pre-encoded), then we cannot use the one coming from the inference outputs, we want to re-generate it from the z coming from the inference outputs. Does it make sense now ?